src/models/edge.ts

```diff
@@ -293,10 +338,105 @@ abstract class Edge<N extends INodeBase, E extends IEdgeBase> implements IEdge<N
         return null;
     }
   }
+
+  setData(data: E): void;
+
+  setData(callback: (edge: IEdge<N, E>) => E): void;
+
+  setData(arg: E | ((edge: IEdge<N, E>) => E)) {
+    if (typeof arg === 'function') {
+      this._data = (arg as (edge: IEdge<N, E>) => E)(this);
+    } else {
+      this._data = arg as E;
+    }
+    this.notifyListeners();
+  }
+
+  patchData(data: Partial<E>): void;
+
+  patchData(callback: (edge: IEdge<N, E>) => Partial<E>): void;
+
+  patchData(arg: Partial<E> | ((edge: IEdge<N, E>) => Partial<E>)) {
+    let data: Partial<E>;
+
+    if (typeof arg === 'function') {
+      data = (arg as (edge: IEdge<N, E>) => Partial<E>)(this);
+    } else {
+      data = arg as Partial<E>;
+    }
+
+    Object.keys(data).forEach((key) => {
+      // @ts-ignore
+      this._data[key] = data[key];
+    });
```

Comment:
Can you try to do this without `// @ts-ignore`. You can read more about why you need to do that - it is because of `Object.keys` - (try sending an object where keys are numbers and see what happens). JS is a weird thing :D

Reply:
Also try to skip using `forEach`. It is ok on simple usages like this which won't be called frequently, but `forEach`, even though the code is nicer, it is slower than `for` or `for in` or `for of`.